### DISCUSSION TITLE
[Question] .editorconfig and .ruleset, which sould I take?

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/roslyn#47580] [General] [answered]
Hello,

I have a (simple) question.
In my projects, I currently use both `.editorconfig` and `.ruleset`, but I have seen I can set the code-rules in the `.editorconfig` too.
In the longer run it is better to start to switch completely to the `.editorconfig` and avoid use of `.ruleset`?

Thanks for you attention

**Comment by PathogenDavid** [comment, marked as answer]
[From the documentation](https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019#convert-an-existing-ruleset-file-to-editorconfig-file):

> Starting in Visual Studio 2019 version 16.5, ruleset files are deprecated in favor of the EditorConfig file for analyzer configuration for managed code.
